### DISCUSSION TITLE
SMTP server password

### DISCUSSION THREAD

**Discussion by doncicuto** [zitadel/zitadel#6378] [Contributors]
One question about the UI that maybe @juergrinaldi and @peintnermax can answer it.

Right now, when I set the SMTP config for my instance, the password must be set once the rest of the config is added before and after that I've to use a new modal/popup. Sometimes as a user I'm tempted to click in Save once the password has been already saved and get the settings haven't been changed message :smile: 

In other apps I usually add the password in the same step as I add the SMTP configuration (server, port, user...), so I'd like to ask you if it would be fine to remove that modal or if it's like this for security / API reasons.

Thanks in advance

**Comment by peintnermax** [comment]
Good point.  👍 We decided to go with a button as we have a seperate request to change the password (`updateSMTPConfigPassword`).
We could remove the button and add an additional field for the password if no config ist set and show the button once some data is set. Keeping the field empty for an update could confuse users as the field is visible and empty. 
An additional approach could be to have a checkbox stating `update password` which enables an additional password field for the update. We would have to add a second api call with this option. What do you think?

**Comment by doncicuto** [comment]
Thanks @peintnermax, I like your approach. That way we can directly show the password field for initial configuration and keep the button for password updates. 

If you agree I'll add an issue explaining this change and prepare a PR :wink:

**Reply by peintnermax** [reply]
That would be great! Thanks for your work @doncicuto 👍
